In [23]:
from IPython.display import HTML

HTML('''<script>
code_show=true; 
function code_toggle() {
 if (code_show){
 $('div.input').hide();
 } else {
 $('div.input').show();
 }
 code_show = !code_show
} 
$( document ).ready(code_toggle);
</script>
<form action="javascript:code_toggle()"><input type="submit" value="Click here to toggle on/off the raw code."></form>''')

In [32]:
import matplotlib.pyplot as plt
from matplotlib import style
import numpy as np
import pandas as pd
import ipywidgets as widgets
from scipy import special
from pandas import Series
import plotly as py
import plotly.graph_objs as go
import math
from IPython.display import Image

py.offline.init_notebook_mode(connected=True)

<font style = 'font-family:Book Antiqua'><i><font size = 4>Thinkful Data Science Boot Camp: Unit 1.5 Capstone</i>
<p></p>
    <p style='line-height:1'><font size = 8><b> U.S.A. Minimum Wage and Median Home Price</b></font></p>

<font style = 'font-family:Book Antiqua'><font size = 6><b><u>The Background</u></b>

<font style = 'font-family:Book  Antiqua'><font size = 5><i><p><b>Exploring Minimum Wages by State from 1968 to 2017</b></i>

<font style = 'font-family:Book Antiqua'><font size = 4><p>To better analyze the minimum wage, lets consolidate all the states and the Federal Minimum wage to one mean value for each year. The chart preview below calculates the mean instance for the values <i>High_Value, Low_Value, CPI_Average, High_2018</i>, and <i>Low_ 2018</i> for all states and D.C. then groups them by year. Note that the value for <i>CPI_Average</i> remains the same because the value displayed in the original data set was already the average calculated value for that given year, repeated every year for each state.</p>

In [24]:
# Importing file with wage data
wage = pd.read_csv('Minimum Wage Data.csv', encoding = "Windows-1252" )

# Converting the State column from series to string data
wage['State'].to_string

# Removing all rows that do not contain the 50 U.S. States but leaving the Federal minimum wage (FLSA).
wage = wage[~wage['State'].isin(['Guam','District of Columbia','Puerto Rico','U.S. Virgin Islands'])]

# Dropping Footnote and Table_Data column from data set.
wage = wage.drop(columns = ['Footnote','Table_Data'])


# Computing the yearly mean wage each year in the United States.
mean_yearly_wage = wage.groupby(['Year'], as_index=False).mean()
mean_yearly_wage.head()

Year  High_Value  Low_Value  CPI_Average  High_2018  Low_2018
0  1968    0.889534   0.834299    34.783333   6.404510  6.006863
1  1969    0.889534   0.834299    36.683333   6.072157  5.695490
2  1970    0.972745   0.917020    38.825000   6.273137  5.913333
3  1971    0.972745   0.917020    40.491667   6.014314  5.669804
4  1972    1.171765   1.109176    41.816667   7.016471  6.641569

<p><font style = 'font-family:Book Antiqua'><font size = 4>The graph below displays the mean wage over time, rounded to the closest penny. We can see that wage has increased over time from \$.89 cents in 1968 to \$7.50 in 2017, which is \$.25 higher than the <a href='https://www.dol.gov/whd/minimumwage.htm'> current Federal Minimum Wage</a>. It is likely that this discrepency is due to the high minimum wage in states like Washington, even though some states do not report their own minimum wage.

<p>The year 1991 sees a jump in mean wage from the steady increase over time. This is likely due to the high increase of the minimum wage in Washington, Oregon, and Alaska that happened that same year (see graph <i>Wage Over Time: by State and Year</i>).</p> 

<p>Though the Federal Minimum Wage has not increased since 2009, we can see that the mean wage in the U.S. has continued to increase. This is likely in relation to states adapting their own minimum wage that is higher than the federal mandated one. In the graph <i>Wage Over Time: by State and Year</i>, we can see that most of the states are clusterd closer together in the earlier years between 1968-1975 and 1980-1990 and as wages begin to vary and increase, so does the mean wage in the U.S.</p>


<p>We can compare the mean wage in each given year to the 2018 equivalent wage, <i>High_2018</i>. The value of any given wage if it was earned in 2018. So the wage of \$0.89 cents is is the 2018 equivalant of \$6.40.</p>

<p>Note that over time the mean minimum wage and the 2018 equivalent get closer together as wages no longer need to be adjusted for inflation. As we have seen on the above graphs, the minimum wage increases at a steady pace with some short periods of no growth. Looking at the <i>2018 Wage Equivalent</i> overlayed, we can see that there are some points on the line where the 2018 wage equivalent drops down closer to the real wage.</p>

<p>Generally, the declines in equivalent value seems to coincide with a <a href='https://www.thebalance.com/the-history-of-recessions-in-the-united-states-3306011'> recession</a>, with the exception of 2008's <a href='https://www.thebalance.com/the-great-recession-of-2008-explanation-with-dates-4056832'>Great Recession</a>. The reason for the drop is probably because the value of the U.S. dollar at those points in time had decreased in value and the purchasing power dropped, even though minimum wages continued to rise. In terms of 2018 dollars, that money did not go as far.</p>

In [25]:
layout = dict(title = 'Mean Wage Over Time: A Comparison To 2018 Value',
              xaxis = dict(title = 'Year'),
              yaxis = dict(title = 'Wage in U.S. $'),
              annotations=[
                dict(
                    x=1973,
                    y=6.61,
                    xref='x',
                    yref='y',
                    text='OPEC Oil Embargo',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=4,
                    ax=0,
                    ay=70
                ),
                dict(
                    x=1980,
                    y=6.37,
                    xref='x',
                    yref='y',
                    text='Interest Rate Hike',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=4,
                    ax=0,
                    ay=-50
                ),             
                dict(
                    x=1990,
                    y=4.96,
                    xref='x',
                    yref='y',
                    text='Savings/Loan Crisis',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=4,
                    ax=0,
                    ay=30
                ),        
                dict(
                    x=2008,
                    y=6.88,
                    xref='x',
                    yref='y',
                    text='Great Recession',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=4,
                    ax=0,
                    ay=-45,
                ),
                dict(
                    x=1970,
                    y=6.27,
                    xref='x',
                    yref='y',
                    text='1970 Recession',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=4,
                    ax=0,
                    ay=-45,
                ),                                       
                dict(
                    x=2001,
                    y=6.25,
                    xref='x',
                    yref='y',
                    text='Dot-Com Crash',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=4,
                    ax=0,
                    ay=-45,
                ),                         
              ]            
        )


# Create a trace
trace5 = go.Scatter(
     x = mean_yearly_wage['Year'],
     y = round(mean_yearly_wage['High_Value'],2),
     name = 'High Yearly Wage',
     line = dict(
         color = ('rgb(102, 183, 163)')
     )
)     

trace6 = go.Scatter(
     x = mean_yearly_wage['Year'],
     y = round(mean_yearly_wage['High_2018'],2),
     name = '2018 Wage Equivelant',
     line = dict(
         color = ('rgb(184, 232, 16)')
     )
)

data = [trace5,trace6]

fig = dict(data=data, layout=layout)
py.offline.iplot(fig, filename='styled-line')

<font style = 'font-family:Book  Antiqua'><font size = 5><i><b>Exploring Median Home Prices Between 1963 to 2011</b></i>
    

<br><font style = 'font-family:Book Antiqua'><font size = 4><p>The <a href='https://www.census.gov/construction/nrs/historical_data/index.html'>United States Census Bureau</a> keeps records of home sales across the country. The below data set is for  new, single-family homes only from 1963 to 2011. The Survey of Construction does not collect sales information for multifamily buildings or for existing homes. By taking the yearly median home value we can better analyze the the trends in home prices.</p>

In [34]:
# Importing file with median price of new homes.
median_home = pd.read_csv('median_home_price.csv')
median_home['Adjusted Home'] = median_home['Median_Home_Price']/1000

# Computing the yearly mean wage each year in the United States.
mean_yearly_home = median_home.groupby(['Year'], as_index = False).mean()
mean_yearly_home.head()

Year  Median_Home_Price  Adjusted Home
0  1963       17966.666667      17.966667
1  1964       18983.333333      18.983333
2  1965       20150.000000      20.150000
3  1966       21525.000000      21.525000
4  1967       22691.666667      22.691667

<br><font style = 'font-family:Book Antiqua'><font size = 4><p>In the below graph CPI and home prices are again displayed, however, the slope of home prices is more discernable. In this graph we can see that home prices actually stabalized in 2006, two years before the housing bubble burst. We can see that 2009 was the lowest point in value for new, single-family homes. Again we see that home values are significantly higher than CPI from 2003 to 2008. Though CPI was increasing and seemed healthy, it was not able to predict the housing market crash. This is likely because of the way housing is calculated, where it calculates the monthly cost of owning a home based on rent prices. As we saw during the Great Recession, the number of vacant homes increased, causing the rental prices to also increase as new renters flooded the market. This means that the CPI was given a false low reading during the housing boom, pre-Great Recession.</p>

In [27]:
layout = dict(title = 'Yearly Median Home Value (<i>New, Single-Family Homes</i>) and CPI Over Time',
              xaxis = dict(title = 'Year'),
              yaxis = dict(title = '$ U.S. '),
            annotations=[
                dict(
                    x=2006,
                    y=244,
                    xref='x',
                    yref='y',
                    text='Home prices stabalize',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=1,
                    ax=10,
                    ay=-70
                ),
                dict(
                    x=2009,
                    y=213,
                    xref='x',
                    yref='y',
                    text='Lowest home values',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=1,
                    ax=10,
                    ay=70
                ),
            ]
    )


# Create a trace
trace7 = go.Scatter(
     x = mean_yearly_wage['Year'],
     y = mean_yearly_wage['CPI_Average'],
     name = 'Average CPI',
     line = dict(
         color = ('rgb(102, 183, 163)')
     )
)     

trace8 = go.Scatter(
     x = mean_yearly_home['Year'],
     y = round(mean_yearly_home['Adjusted Home'],2),
     name = 'Home Price Displayed in $1000s',
     line = dict(
         color = 'red'
     )
)

data = [trace7,trace8]

fig = dict(data=data, layout=layout)
py.offline.iplot(fig, filename='styled-line')


<font style = 'font-family:Book Antiqua'><font size = 5><i><b>Comparing Wage and Median Home Prices</b></i>

<br><font style = 'font-family:Book Antiqua'><font size = 4><p>By combining the created data sets <i>Median Wage</i> and <i>Median Home Prices</i> we can create two new columns with the percent change of values from <i>High_Value</i>(median minimum wage in a given year) and <i>Median_Home_Price</i> (new, single-family home prices). See columns in data set below <i>PercentHome_Change</i> and <i>PercentWage_Change</i></p>
<br>
<b><u>Added Chart Values:</u></b>
<p class='tab'><p>PercentHome_Change: Percent change from previous year for median minimum wage.
<p>PercentWage_Change: Percent change from previous year for median prices of new, single-family homes.</p> 

In [30]:
# Merging both data sets wage and median home.
wage_and_medianhome = mean_yearly_wage.merge(mean_yearly_home, on = 'Year')


wage_and_medianhome['PercentHome_Change'] = wage_and_medianhome['Median_Home_Price'].pct_change(1)
wage_and_medianhome['PercentWage_Change'] = wage_and_medianhome['High_Value'].pct_change(1)

wage_and_medianhome.head()

Year  High_Value  Low_Value  CPI_Average  High_2018  Low_2018  \
0  1968    0.889534   0.834299    34.783333   6.404510  6.006863   
1  1969    0.889534   0.834299    36.683333   6.072157  5.695490   
2  1970    0.972745   0.917020    38.825000   6.273137  5.913333   
3  1971    0.972745   0.917020    40.491667   6.014314  5.669804   
4  1972    1.171765   1.109176    41.816667   7.016471  6.641569   

   Median_Home_Price  Adjusted Home  PercentHome_Change  PercentWage_Change  
0       24833.333333      24.833333                 NaN                 NaN  
1       25575.000000      25.575000            0.029866            0.000000  
2       23533.333333      23.533333           -0.079831            0.093544  
3       25216.666667      25.216667            0.071530            0.000000  
4       27550.000000      27.550000            0.092531            0.204596

<br><font style = 'font-family:Book Antiqua'><font size = 4><p>The graph below, <i>Minimum Wage and Median Single-Family Home Price: Yearly Percent Change</i>, is a better visual representation of the changes in each year. Here, we can compare the changes in wages and homes each year. As we can see, percent wage change varies over the years, but never dips below zero. This supports the continual upward trend seen in graph <i>Wage Over Time: By Year</i>. In contrast, percent change for new, single-family home values dip below zero on four seperate occasions. These occasions align with the 1970's Recession, savings/loan crisis, and the Great Recession.</p> 
<br>
<p>Notably, there seems to be greater increase in wages before 1995. After 1995, the changes are more moderate with the majority being less than a .1% change. At the same time, home values seem to be increasing at a higher percent than wages. Between the years 2001 and 2005, home values increased by at least twice or more percentage points than wages. Then the Great Recession happened and home prices plummetted before increasing again in 2010 and 2011.</p>

In [33]:
layout = dict(title = 'Minimum Wage and Median Single-Family Home Prices: Yearly Percent Change',
              xaxis = dict(title = 'Year'),
              yaxis = dict(title = 'Percent Change')
             )
 
data = [
    go.Bar(
        x = wage_and_medianhome['Year'],
        y = round(wage_and_medianhome['PercentHome_Change'],2),
        base = 0,
        marker = dict(
          color = ('rgb(184, 232, 16)')
        ),
        name = '% Home Price Change'
    ),
    go.Bar(
        x = wage_and_medianhome['Year'],
        y = round(wage_and_medianhome['PercentWage_Change'],2),
        base = 0,
        marker = dict(
          color = ('rgb(102, 183, 163)')
        ),
        name = '% Wage Change'
    )
]


fig = go.Figure(layout=layout,data=data)
py.offline.iplot(fig, filename='base-bar')

<font style = 'font-family:Book Antiqua'><font size = 4><p> Overall we can see that wages and new, single-family home prices have a similar trend of continual growth, despite some moments of decline. Further research is needed to truly assess if home values are increasing at a greater rate then wages. It would be beneficial to examine home prices by state rather than month. It might also be interesting to examine home prices for all types of homes, rather than just new, single-family homes as we did in this analysis. It would be intresting to note how each state varied in regards to its wages and home prices.</p>

<hr width=50%>

<font style = 'font-family:Book Antiqua'><font size = 4><i>Experimental Design:</i>
<p></p>
    <p style='line-height:1'><font size = 8><b> Will Increased Minimum Wage in the Most Expensive Cities in the USA Improve Home Affordability?</b></font></p>


<font style = 'font-family:Book Antiqua'><font size = 4><p> According to the article published by Dan Burrows, <i><a href='https://www.kiplinger.com/slideshow/real-estate/T006-S001-most-expensive-u-s-cities-to-live-in-2019/index.html'>20  Most Expensive U.S. Cities to Live In</a></i>, the cities of Honolulu, Seattle and Manhattan are the top three most expensive cities to live in the United States with a median home value of \\$626k to \\$927k. Though most of these cities do have a higher wage than the federal minimum, Seattle and Manhattan as high as \\$12.</p>
<br>
<p>Through the analysis above we can see that overall, the wage in the United States has increased through the years, the data cannot truly represent what is happening in this cities with high priced homes and high cost of living. For example, Manahattan's median home price is \$915,300 but Albany, a county north of Manhattan, has a median home price of <a href='https://www.tax.ny.gov/research/property/assess/sales/resmedian.htm'>\$210,000</a>. Large distributions among two different counties in the same state can create a skewed picture of median home prices in the whole state.</p>
<br>
<p>In the experiment below, we will focus on these high price cities and see if a wage increase will make homeownership, a pillar of the American Dream, come true for more Americans in those cities.</p>

<font style = 'font-family:Book Antiqua'><font size = 6><p><b><u>Experimental Design</u></b></p>

<font style = 'font-family:Book  Antiqua'><font size = 5><i><b>Hypothesis:</b></i>
<font style = 'font-family:Book  Antiqua'><font size = 4><p>Will an increase in minimum wage to \$20 an hour in the cities of Honolulu, Hawaii; Seattle, Washington and Manhattan, New York improve home affordability for low-income families?

<font style = 'font-family:Book  Antiqua'><font size = 5><i><b>Sample:</b></i>
<font style = 'font-family:Book  Antiqua'><font size = 4><p>We will focus on three major cities:
- Honolulu, Hawaii
- Seattle, Washington
- Manhattan, New York

<font style = 'font-family:Book  Antiqua'><font size = 4>Our analysis will also focus one the affects of a wage increase of families that make 80 percent or less of Area Median Income (AMI):

<b><a href='https://dbedt.hawaii.gov/hhfdc/files/2018/04/2018-HUD-Income-Limits-All-Counties.pdf'>Honolulu, Hawaii:</a></b>

<img src=Honolulu.jpg>

<b><a href='https://www.seattle.gov/Documents/Departments/Housing/PropertyManagers/IncomeRentLimits/2018%20Rent%20and%20Income%20Limits.pdf'>Seattle, Washington:</a></b>
<img src=Seattle.jpg>

<b><a href='https://www1.nyc.gov/site/hpd/renters/area-median-income.page'>Manhattan, New York:</a></b>
<img src=NewYork.jpg>

<font style = 'font-family:Book Antiqua'><font size = 6><b><u>
    Rollout & Evaluation Plan</u></b>

<font style = 'font-family:Book Antiqua'><font size = 5><i><b>Impact Window:</b></i> 

<font style = 'font-family:Book Antiqua'><font size = 4>The first 5 years after the minimum wage is increased to \\$20 an hour.

<font style = 'font-family:Book Antiqua'><font size = 5><i><b>Success Metric:</b></i>

<font style = 'font-family:Book Antiqua'><font size = 4>An increase of 2% in home ownership.
Increased home ownership in the cities of Honolulu, Seattle and San Francisco among households that make 80\% percent or less of AMI.

<font style = 'font-family:Book Antiqua'><font size = 5><i><b>Secondary Metrics:</b></i> 

<font style = 'font-family:Book Antiqua'><font size = 4>There are other metrics that can be viewed to measure the success of the wage increase:    
- Number of home foreclosures.
- Amount of families lifted out of poverty.
 - This can be counted by the number of households that no longer rely on the following programs:
   - Supplemental Nutrition Assistance Program (SNAP)
   - The Special Supplemental Nutrition Program for Women, Infants, and Children (WIC)
   - Temporary Assistance for Needy Families (TANF)
   - Medicaid
   - Housing Choice Voucher Program/Section 8 Housing

<font style = 'font-family:Book Antiqua'><font size = 5><i><b>Data Collection Approach:</b></i>

<font style = 'font-family:Book Antiqua'><font size = 4>To measure the affects of a wage increase on any of the metrics mentioned above, we use state and federal records to gather data and make observations about the population we have chosen to focus on. Specifically, we will use data from <a href='https://www.census.gov/programs-surveys/ahs.html'>American Housing Survey (AHS)</a> to calculate a change in home ownership for Manhattan and Seattle. Obtaining housing information for Honolulu may be more difficult as Hawaii does not participate in AHS surveys. Because our research will focus more on the low income population it might be better to narrow our scope and use data sets from HUD. The Resident Characteristics Report is released once a month and summarizes general information about households who reside in Public Housing or participate in Section 8. Obtaining data to measure public assistance programs may be more difficult as most data is gathered by county or at the state level and rarely by metropolitian area. However, if the data allowed, we can break down the information by zip code and focus on those within Honolulu, Seattle, and Manhattan.

<font style = 'font-family:Book Antiqua'><font size = 4>There are a number of limitaions in this study:
- The study assumes that all those who work in the the cities of Honolulu, Seattle, and Manhattan will also reside in those cities. This study there for does not take into account the residency of all workers in the city.
- This analysis does not take into account fluctiation of home prices and rents nor the effects a wage incresae could have on both.
- It is likely that with the increase among low-income earners, those in more senior and mamagement roles can also see a change in their owm waged as they increase to maintain pay scale. However this "wage spill" will not be reviewed in this paper.
- The time frame of this study can present problems and create variances within the data. A recession can cause higher unemployement and increase the popuation of households receiving government assistance as well as increase mortgage defaults, decreasing homeowenership rates. 

<font style = 'font-family:Book Antiqua'><font size = 5><i><b>Success Criteria:</b></i>

<font style = 'font-family:Book Antiqua'><font size = 4>I would not consider any increase in homeownership a failure. A 2% increase is aggressive, especially considering the relative short time frame of 5 years. When looking at anything related to policy changes, it will take time to see real changes. Though agressive, a 2% increase in homeownership is still feasible. Information from the <a href='https://fred.stlouisfed.org/series/RHORUSQ156N#0'>US Census Bureau shows</a> that a 2% increase is possible. The below chart shows a 1.9% increase in homeownership in the United States between 4/1/2016 and 10/1/2018.

In [20]:
homeownership = pd.read_csv('RHORUSQ156N (1).csv')
homeownership.head()

layout = dict(title = 'U.S. Homeownership Over Time',
              xaxis = dict(title = 'Year'), 
              yaxis = dict(title = 'Homeownership %'),
            annotations=[
                dict(
                    x='4/1/2016',
                    y=63.4,
                    xref='x',
                    yref='y',
                    text='62.9%',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=3,
                    ax=-10,
                    ay=-80
                ),
                dict(
                    x='10/1/2018',
                    y=64.8,
                    xref='x',
                    yref='y',
                    text='64.8%',
                    showarrow=True,
                    bgcolor= 'yellow',
                    arrowhead=3,
                    ax=10,
                    ay=-70
                ),
            ]
    )


# Create a trace
trace13 = go.Scatter(
     x = homeownership['DATE'],
     y = homeownership['Home_Ownership'],
     name = 'Average CPI',
     line = dict(
         color = ('rgb(102, 183, 163)')
     )
)     


data = [trace13]

fig = dict(data=data, layout=layout)
py.offline.iplot(fig, filename='styled-line')

<font style = 'font-family:Book Antiqua'><font size = 4>If there is no increase in homeownership, there are other ways in which the success of a minimum wage increase can be measured. An overall decrease in the level of households living at or below 80% of AMI and a decrease in households relying on government assistance programs can be considered a success. Historically, owning a home has become one of the American milestones by which success is measured but maybe this view is dated and should change. The majority of a households income is in the value of their home, however this is not the only investment that is available. If an increase in wages can lift households out of poverty and give families more disposable income, they may be more likely to take part in other investment avenues that can increase their net worth.